Source/Core/VideoBackends/Vulkan/StateTracker.cpp

```diff
@@ -322,44 +338,134 @@ bool StateTracker::CheckForShaderChanges(u32 gx_primitive_type)
 {
   VertexShaderUid vs_uid = GetVertexShaderUid();
```

Comment:
Just a couple of questions about this function:  1) Why does this return a bool if it never uses it?  2) Where is this logic captured in the other backends?  Is there any particular reason this is so different?